### DISCUSSION TITLE
No PUT in API [Apr 26]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#1960] [Discord Threads]
> 🙋‍♂️ **Zorphix**

**Comment by o1lab** [comment]
> 👨‍🚀 **wingkwong**  
#### PUT would overwrite the entire entity for a existing record, while PATCH only update partially. In nocodb, we only update those fields user've changed, hence the API method s chosen as PATCH.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/967889400853844018)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)

**Reply by isafiur** [reply]
Hi
I used PUT to Create/Update records to tables. Now with PATCH, the update is working well. However I couldn't figure out how to create new record in nocoDB

**Reply by wingkwong** [reply]
@isafiur you should use `POST` for creating a new record.

**Reply by isafiur** [reply]
I have been using xmysql for some time and it was easy there. Now I have to edit every where I have used "PUT" and execute "GET" to know the status. Then I have to execute "POST" if record not found and "PATCH" if record is already there.
Is there any easy method present in nocodb which can be alternative to PUT ?
